File: Crowmask.Interfaces/IRequest.vb
Imports System.Net.Http
Imports System.Net.Http.Headers

''' <summary>
''' An abstraction of request data used to validate incoming HTTP signatures.
''' </summary>
Public Interface IRequest
    ''' <summary>
    ''' The HTTP method (e.g. GET, POST).
    ''' </summary>
    ''' <returns></returns>
    ReadOnly Property Method As HttpMethod

    ''' <summary>
    ''' The request URI (e.g. https://crowmask.example.com/api/actor/inbox).
    ''' </summary>
    ''' <returns></returns>
    ReadOnly Property RequestUri As Uri

    ''' <summary>
    ''' A collection of request headers.
    ''' </summary>
    ''' <returns></returns>
    ReadOnly Property Headers As HttpHeaders
End Interface

File: Crowmask.HighLevel/Signatures/IRequest.cs
using System.Net.Http.Headers;

namespace Crowmask.HighLevel.Signatures
{
    /// <summary>
    /// An abstraction of request data used to validate incoming HTTP signatures.
    /// </summary>
    public interface IRequest
    {
        /// <summary>
        /// The HTTP method (e.g. GET, POST).
        /// </summary>
        HttpMethod Method { get; }

        /// <summary>
        /// The request URI (e.g. https://crowmask.example.com/api/actor/inbox).
        /// </summary>
        Uri RequestUri { get; }

        /// <summary>
        /// A collection of request headers.
        /// </summary>
        HttpHeaders Headers { get; }
    }
}
